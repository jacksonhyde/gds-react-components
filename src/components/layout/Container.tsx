import React, { HTMLProps } from 'react';
import classNames from 'classnames';

interface ContainerProps extends HTMLProps<HTMLDivElement> {
  fluid?: boolean;
}

const Container: React.FC<ContainerProps> = ({ className, fluid, ...rest }) => (
  <div
    className={classNames(
      { 'govuk-width-container': !fluid },
      { 'govuk-width-container-fluid': fluid },
      className,
    )}
    {...rest}
  />
);

export default Container;
