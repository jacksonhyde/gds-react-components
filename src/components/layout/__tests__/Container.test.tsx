import React from 'react';
import { shallow } from 'enzyme';
import Container from '../Container';

describe('Container', () => {
  it('matches snapshot', () => {
    const component = shallow(<Container />);
    expect(component.hasClass('govuk-width-container-fluid')).toBeFalsy();
    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it('adds fluid classes', () => {
    const component = shallow(<Container fluid />);
    expect(component.hasClass('govuk-width-container-fluid')).toBeTruthy();
    component.unmount();
  });
});
