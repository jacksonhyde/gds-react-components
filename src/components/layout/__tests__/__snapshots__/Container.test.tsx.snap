// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container matches snapshot 1`] = `
<div
  className="govuk-width-container"
/>
`;
